### PR TITLE
match project metadata popup style with project statistics popup style

### PR DESCRIPTION
**Pull request by lennartclaas**
### Summary of changes

The project metadata popup table style was adjusted according to the project statistics popup table style. 

### Context and reason for change

Before the changes, the project metadata popup table style was quite plain. Now, both tables have the same style, which improves visual consistency.

### How can the changes be tested

Open both tables with the "File" button and take a look.

### Comments

Added maxWidth to NotificationPopupProps in order to enable different widths for the project metadata popup (use value 'sm') and the project statistics popup (use value 'xl').
projectMetadataPopupClasses were defined to remove any dependencies from the project statistics popup style. Later on, both popup tables can depend on a single style (other ticket).

Before:
0_opossumui_project_metadata_old

After:
3_opossumui_project_metadata_final

## UPDATE
The width of the project metadata popup is now flexible and adapts to the table contents.  Suggestions regarding code cosmetics have been applied.

small table example:
0_opossumui_project_metadata_adaptable_small

large table example:
0_opossumui_project_metadata_adaptable_large

Issue: #1010

### CODE REVIEW

**Comment by vasily-pozdnyakov** [comment]
It would be nice to add screenshots of the component before and after the changes.

**Comment by vasily-pozdnyakov** [comment]
The popup is not flexible, if you open it for a bigger file (e.g. opossum_ui_scan_input.json.gz) with a lot of content:
image

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx [hunk 1]
```diff
@@ -5,79 +5,29 @@
...
-      content={content}
+      content={
+        <>
+          <ProjectMetadataTable />
+        </>
+      }
       header={'Project Metadata'}
       isOpen={true}
       fullWidth={true}
+      maxWidth="sm"
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx:30]: why was this changed?

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx [hunk 2]
```diff
@@ -5,79 +5,29 @@
...
-        </MuiTableBody>
-      </MuiTable>
-    </MuiTableContainer>
-  );
-
   return (
     <NotificationPopup
-      content={content}
+      content={
+        <>
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx:23]: the fragment `<>`is not needed

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:1]: This component should have its own folder directly under `components/`

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { ReactElement } from 'react';
+import MuiBox from '@mui/material/Box';
+import MuiTable from '@mui/material/Table';
+import MuiTableBody from '@mui/material/TableBody';
+import MuiTableCell from '@mui/material/TableCell';
+import MuiTableContainer from '@mui/material/TableContainer';
+import MuiTableRow from '@mui/material/TableRow';
+import { projectMetadataPopupClasses } from './../ProjectMetadataPopup/shared-project-metadata-popup-styles';
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:13]: these styles seem not shared and should be moved into this file. Even Though I understand that @FlorianSchepers might also touch the styling. Not sure how to coordinate here best.

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import { ProjectMetadata } from '../../../shared/shared-types';
+import { getProjectMetadata } from '../../state/selectors/all-views-resource-selectors';
+import { useAppSelector } from '../../state/hooks';
+import MuiTypography from '@mui/material/Typography';
+
+export function ProjectMetadataTable(): ReactElement {
+  const metadata: ProjectMetadata = useAppSelector(getProjectMetadata);
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:20]: The type does not have to be provided here

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  function mapKeyToString(key: string): string {
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:22]: `mapKeyToString` -> `mapKeyToDisplayName`

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    const keyToDisplayName: { [key: string]: string } = {
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:23]: move to beginning of file

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      projectTitle: 'Project Title',
+      projectId: 'Project ID',
+      fileCreationDate: 'File Creation Date',
+    };
+
+    if (key in keyToDisplayName) {
+      return keyToDisplayName[key];
+    } else {
+      return key;
+    }
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:33]: might be `return key in keyToDisplayName ? keyToDisplayName[key] : key`

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  function getMuiTableRow(key: string): ReactElement {
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:36]: `getMuiTableRow` -> `getTableRow`

The prefix "Mui" we only use for components imported from material-ui

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            </MuiTypography>
+          </pre>
+        </MuiTableCell>
+      </MuiTableRow>
+    );
+  }
+
+  return (
+    <MuiBox>
+      {/* <MuiTypography variant="subtitle1">{'Test'}</MuiTypography> */}
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:57]: remove

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx (hunk 5 above)
Comment (benedikt-richter) [reply src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataTable.tsx:20]: also metadata is verty generic. Might be called `projectMetadata`

File: src/Frontend/Components/ProjectMetadataPopup/shared-project-metadata-popup-styles.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
```
Comment (benedikt-richter) [inline src/Frontend/Components/ProjectMetadataPopup/shared-project-metadata-popup-styles.ts:2]: There is no need for a file containing shared styles here, see above.

**Review by benedikt-richter — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx (hunk 1 above)
Comment (benedikt-richter) [reply src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx:30]: It seems this change leads to the behavior documented in https://github.com/opossum-tool/OpossumUI/pull/1185#issuecomment-1326672960

Comment (lennartclaas) [reply src/Frontend/Components/ProjectMetadataPopup/ProjectMetadataPopup.tsx:30]: Previously, maxWidth was hardcoded with value 'xl'. This led to smaller tables looking like this:
2_opossumui_project_metadata_new_wide

I guess the best solution would be to employ a flexible table, as proposed in [#1185 (comment)](https://github.com/opossum-tool/OpossumUI/pull/1185#issuecomment-1326672960)

**Comment by benedikt-richter** [comment]
Regarding the commit message, I'd like to refer to `https://github.com/opossum-tool/OpossumUI/blob/main/CONTRIBUTING.md`. A good example is given there:

```

Short summary of changes (72 characters or less)

More detailed explanatory text, if necessary. This text can have 
several lines.

Issue: #123  (this line is optional)

```

In case you replace "Issue" by "Fix" the corresponding issue will be automatically closed, when this PR is merged.

**Review by benedikt-richter — APPROVED** [review]
Looks good! 

Nice solution for the issue of various width 👍 

Just rebase on the latest `main` before merging

File: src/Frontend/Components/ProjectMetadataTable/ProjectMetadataTable.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    width: 'max-content',
+    overflow: 'auto',
+    color: OpossumColors.black,
+  },
+  projectMetadataTableContainer: {
+    maxHeight: '500px',
+    width: 'max-content',
+    marginBottom: 3,
+  },
+};
```
Comment (FlorianSchepers) [inline src/Frontend/Components/ProjectMetadataTable/ProjectMetadataTable.tsx:38]: Seems you made some changes to these styles, so a refactoring with my pull request is no longer necessary.

**Review by FlorianSchepers — COMMENTED** [review]
lgtm

**Pull request merged**